syntax = "proto3";


import "google/protobuf/any.proto";
import "google/protobuf/empty.proto";
import "google/protobuf/struct.proto";
import "google/protobuf/timestamp.proto";
import "google/protobuf/wrappers.proto";


message MainMessage {
    string a_str = 1;
    map<int32, ListOfIntsMessage> int_to_lst_ints_map = 2;
    SomeEnum an_enum = 3;
    int32 an_int = 4;
    repeated int32 lst_ints = 5;
    map<string, SomeSubMessage> str_to_message_map = 6;
    map<string, int32> str_to_int_map = 7;
}

message SomeSubMessage{
    string a_str = 1;
    int64 a_long = 2;
    repeated int64 lst_longs = 3;
}

enum SomeEnum {
        first = 0;
        second = 1;
}

message ListOfIntsMessage {
    repeated int32 lst_ints = 1;
}

// Test a message containing a list of a messages which in turn contain a string and a struct.
message MessageWithListOfMessages {
    repeated SomeSubMessage lst_messages = 1;

    // a message containing a string and a struct.
    message SomeSubMessage{
        string a_str = 1;
        google.protobuf.Struct a_struct = 2;
    }
}

